WalletWasabi.Gui/ViewModels/AddressViewModel.cs

```diff
@@ -123,13 +124,20 @@ public AddressViewModel(HdPubKey model, KeyManager km)
 				}
 			});
 
+			LockAddress = ReactiveCommand.Create(() =>
+			{
+				var key = km.GetKeys(x => !x.IsInternal && x == Model).FirstOrDefault();
```

Comment:
```suggestion
				var key = km.GetKeys(x => !x.IsInternal && x == Model).Single();
```

Reply:
https://www.c-sharpcorner.com/blogs/singleordefault-vs-firstordefault-in-linq-query1
https://stackoverflow.com/questions/1745691/linq-when-to-use-singleordefault-vs-firstordefault-with-filtering-criteria

If your suggestion was for better readability then using `Single()`  is better.
If it was for performance then `FirstOrDefault()` performs better (not really significant).